server/utils/user/UserFactory.ts

```
@@ -0,0 +1,213 @@
+        }
+        return new AnonymousUser(uuid(), 'AnonymousPlayer');
+    }
+
+    /**
+     * TODO Change username with rate limiting (30 days) change the rate limit
+     * @param userId The user ID
+     * @param newUsername The new username
+     * @returns Object with success status and message
+     */
+    public async changeUsername(userId: string, newUsername: string): Promise<boolean> {
+        try {
+            const userProfile = await this.dynamoDbService.getUserProfile(userId);
+            if (!userProfile) {
+                return false;
+            }
+            // Check if username was changed recently (within the last 30 days)
+            if (userProfile.usernameSetAt) {
+                const lastChange = new Date(userProfile.usernameSetAt).getTime();
+                const now = Date.now();
+                const daysSinceLastChange = (now - lastChange) / (1000 * 60 * 60 * 24);
+
+                // If changed within the last 30 days, don't allow another change
+                if (daysSinceLastChange < 30) {
+                    const daysRemaining = Math.ceil(30 - daysSinceLastChange);
+                    return false;
+                }
+            }
+
+            // Update username and set the timestamp
+            await this.dynamoDbService.updateUserProfile(userId, {
+                username: newUsername,
+                usernameSetAt: new Date().toISOString()
+            });
+
+            logger.info(`Username for ${userId} changed to ${newUsername}`);
+            return true;
+        } catch (error) {
+            logger.error('Error changing username:', error);
+            return false;
+        }
+    }
+
+    /**
+     * Update user preferences
+     * @param userId The user ID
+     * @param preferences The updated preferences object
+     * @returns True if update was successful
+     */
+    public async updateUserPreferences(userId: string, preferences: Record<string, any>): Promise<boolean> {
+        try {
```

Also here feels like these should just allow exceptions to float up and let the request fail